bin/list-charts

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+GIT_ROOT=$(cd "$(dirname "$0")/.." && pwd)
+
+function usage() {
+    echo "Usage:"
+    echo "list-charts"
+    echo "    -h                      Display usage."
+    echo "    -a                      List only application charts."
+    echo "    -l                      List only library charts."
+    echo "    -n                      List charts by name."
+    echo "    -p                      List charts by path."
+    exit 0
+}
+
+function error() {
+    echo "${1}" 1>&2
+    exit 1
+}
+
+list_application_charts=false
+list_library_charts=false
+
+list_chart_names=false
+
+while getopts ":halnp" opt; do
+    case ${opt} in
+        h )
+            usage
+        ;;
+        a )
+            list_application_charts=true
+        ;;
+        l )
+            list_library_charts=true
+        ;;
+        n )
+            list_chart_names=true
+        ;;
+        p )
+            list_chart_names=false
+        ;;
+        \? )
+            error "Invalid Option: -${OPTARG}"
+        ;;
+    esac
+done
+shift $((OPTIND -1))
+
+if [[ "${list_application_charts}" == false && "${list_library_charts}" == false ]]; then
+    list_application_charts=true
+    list_library_charts=true
+fi
+
+if [[ "${list_application_charts}" == true && "${list_library_charts}" == true ]]; then
+    chart_paths=$(find "${GIT_ROOT}"/charts/ -type f -name 'Chart.yaml')
```

Comment:
Do we maybe want to enforce having only directories with a `Chart.yaml` file?

We could maybe enforce it with using in  the "false false" case for example:

```sh
chart_paths="$(ls -d charts/*/)"
```

We'll only be able to do this in the top directory (to not match for example the templates dir). It will also not be enforced in the PRs, only on the main branch (due to the optimization to only work on the changed charts for a PR).